filters/drop.hpp

```diff
@@ -0,0 +1,173 @@
+#ifndef SERENITY_DROP_FILTER_HPP
+#define SERENITY_DROP_FILTER_HPP
+
+#include <list>
+#include <memory>
+#include <type_traits>
+
+#include "glog/logging.h"
+
+#include "filters/ema.hpp"
+
+#include "messages/serenity.hpp"
+
+#include "serenity/data_utils.hpp"
+#include "serenity/executor_map.hpp"
+#include "serenity/executor_set.hpp"
+#include "serenity/serenity.hpp"
+
+#include "stout/lambda.hpp"
+#include "stout/nothing.hpp"
+#include "stout/option.hpp"
+#include "stout/result.hpp"
+
+namespace mesos {
+namespace serenity {
+
+struct ChangePointDetection {
+  double_t severity;
+};
+
+
+constexpr uint64_t DEFAULT_WINDOW_SIZE = 10;
+constexpr uint64_t DEFAULT_CONTENTION_COOLDOWN = 10;
+constexpr double_t DEFAULT_ABS_THRESHOLD = 0;
+constexpr double_t DEFAULT_RELATIVE_THRESHOLD = 20;
+
+
+/**
+ * State for every Change Point Detector. It enables user to
+ * configure change Point Detectors.
+ * Detectors are also able to adjust these values dynamically.
+ */
+struct ChangePointDetectionState {
+  static ChangePointDetectionState createForNaiveDetector(
+      uint64_t _contentionCooldown,
```

Comment:
What does this mean? What are the accepted values? Mind documenting 'cool down'?